File: round/src/main/java/src/game_logic/Player.java
package src.game_logic;

import src.game_logic.Rank.RANKS;

public class Player {

	private RANKS rank;
	private AdventureDeck advDeck;
	
	public Player() {
		rank = Rank.RANKS.Squire;
		advDeck = new AdventureDeck();
	}
	
	public void increaseLevel() {
		if(rank == Rank.RANKS.Squire) {
			rank = Rank.RANKS.Knight;
		} else if (rank == Rank.RANKS.Knight) {
			rank = Rank.RANKS.Champion;
		}
	}
	
	public void addCards(Card[] cards) {
		for(Card card: cards) {
			advDeck.addCard(card, 1);
		}
	}

	public String hand() {
		return advDeck.toString();
	}

}

File: Unity/QuestOfTheRoundTable/Assets/Scripts/Objects/Player.cs

using System.Collections;

public class Player {

    public Card.Rank Rank { get;set; }
    public ArrayList Hand;

    public Player()
    {
        Rank = Card.Rank.Squire;
        Hand = new ArrayList();
    }
    public void RankUp()
    {
        if(Rank == Card.Rank.Squire)
        {
            Rank = Card.Rank.Knight;
        }else if(Rank == Card.Rank.Knight)
        {
            Rank = Card.Rank.Champion;
        }
    }
    public void AddCard(Card.AdventureCard c)
    {
        Hand.Add(c);
    }
    public string DisplayHand()
    {
        return Hand.ToString();
    }
}
